### PR TITLE
[wasm] Throw PlatformNotSupportedException instead of NotSupportedException when starting threads/waiting on monitors.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/utils/mono-error-internals.h [hunk 1]
```diff
@@ -278,6 +278,11 @@ mono_error_set_cannot_unload_appdomain (MonoError *error, const char *message)
 	mono_error_set_generic_error (error, "System", "CannotUnloadAppDomainException", "%s", message);
 }
 
+static inline void
+mono_error_set_platform_not_supported (MonoError *error, const char *message)
+{
+	mono_error_set_generic_error (error, "System", "PlatformNotSupportedException", "%s", message);
```
Comment (lambdageek) [inline mono/utils/mono-error-internals.h:284]: Linker descriptor?

**Comment by vargaz** [comment]
Will preserve the netcore version later when that infrastructure is set up.

**Comment by lambdageek** [comment]
> Will preserve the netcore version later when that infrastructure is set up.

Does this file not get used: https://github.com/dotnet/runtime/blob/master/src/mono/netcore/System.Private.CoreLib/src/LinkerDescriptor/System.Private.CoreLib.xml

**Comment by vargaz** [comment]
Yes, but there are all kinds of reorgs happening wrt that code.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@vargaz adding this to the netcore linker xml should be fine now, not sure which reorgs you mean that would affect this.

**Pull request merged**